### DISCUSSION TITLE
Lookup control

### DISCUSSION THREAD

**Discussion by massimotomasi** [Megabit/Blazorise#2971] [Q&A]
I'm trying to develop a Lookup component inheriting from your Autocomplete component.
Basically what i want to do is an Autocomplete that search data using a service instead of searching in local Data collection.

1. I create an interface for the service:

```
  public interface ILookupSearchService<TItem>
  {
      Task<IEnumerable<TItem>> SearchAsync(string searchedText, CancellationToken cancellationToken);
  }

```
2. I used it inside my Lookup control
    
```
  public partial class Lookup<TItem, TValue> : Autocomplete<TItem, TValue>
  {
      private CancellationTokenSource _cancellationTokenSource;
      
      [Parameter]
      public ILookupSearchService<TItem> SearchService{ get; set; }

      protected async Task OnSearchChanged(string text)
      {
          _cancellationTokenSource?.Cancel();
          _cancellationTokenSource = new CancellationTokenSource();

          var data = await SearchService.SearchAsync(text, _cancellationTokenSource.Token);
          Data = data;
      }
  }

```
3. The service implementation uses a http client to load data (from my backend api)
4.  I use the component inside my pages like this (ProductLookupSearchService implements ILookupSearchService<Product>):

```

Lookup TItem="Product"
        TValue="int"
        ValueField="(x) => x.Codice"
        SearchService="ProductLookupSearchService">
    <NotFoundContent>No Products found...</NotFoundContent>
</Lookup>
```

The service is correctly called and data are retried but i can't see those in the drop down menu. What I'm doing wrong?

**Comment by David-Moreira** [comment]
Hello,
I think you might be missing `TextField`. 

https://blazorise.com/docs/extensions/autocomplete/

**Reply by massimotomasi** [reply]
Thx for the reply. For this use case i prefered using Blazored.Typeahead component that has support for this scenario

**Reply by David-Moreira** [reply]
No problem. We'll take that as feedback to improve our component even further, maybe consider support for an inbuilt data provider function on AutoComplete.

**Reply by massimotomasi** [reply]
That would be great
